modules/layers/src/solid-polygon-layer/polygon-tesselator.js

```diff
@@ -160,74 +156,42 @@ function calculateIndices({polygons, IndexType = Uint32Array}) {
 }
 
 function updatePositions({
-  cache: {positions, positions64xyLow, nextPositions, nextPositions64xyLow},
+  cache: {positions, positions64xyLow, vertexEnabled},
   polygons,
   extruded,
   fp64
 }) {
   // Flatten out all the vertices of all the sub subPolygons
   let i = 0;
-  let nextI = 0;
-  let startVertex = null;
-
-  const pushStartVertex = (x, y, z, xLow, yLow) => {
-    if (extruded) {
-      // Save first vertex for setting nextPositions at the end of the loop
-      startVertex = {x, y, z, xLow, yLow};
-    }
-  };
-
-  const popStartVertex = () => {
-    if (startVertex) {
-      nextPositions[nextI * 3] = startVertex.x;
-      nextPositions[nextI * 3 + 1] = startVertex.y;
-      nextPositions[nextI * 3 + 2] = startVertex.z;
-      if (fp64) {
-        nextPositions64xyLow[nextI * 2] = startVertex.xLow;
-        nextPositions64xyLow[nextI * 2 + 1] = startVertex.yLow;
-      }
-      nextI++;
-    }
-    startVertex = null;
-  };
-
   polygons.forEach((polygon, polygonIndex) => {
-    Polygon.forEachVertex(polygon, (vertex, vertexIndex) => {
-      // eslint-disable-line
-      const x = vertex[0];
-      const y = vertex[1];
-      const z = vertex[2] || 0;
-      let xLow;
-      let yLow;
-
-      positions[i * 3] = x;
-      positions[i * 3 + 1] = y;
-      positions[i * 3 + 2] = z;
-      if (fp64) {
-        xLow = fp64LowPart(x);
-        yLow = fp64LowPart(y);
-        positions64xyLow[i * 2] = xLow;
-        positions64xyLow[i * 2 + 1] = yLow;
-      }
-      i++;
-
-      if (extruded && vertexIndex > 0) {
-        nextPositions[nextI * 3] = x;
-        nextPositions[nextI * 3 + 1] = y;
-        nextPositions[nextI * 3 + 2] = z;
+    polygon.forEach(loop => {
+      loop.forEach((vertex, vertexIndex) => {
+        // eslint-disable-line
+        const x = vertex[0];
+        const y = vertex[1];
+        const z = vertex[2] || 0;
+
+        positions[i * 3] = x;
+        positions[i * 3 + 1] = y;
+        positions[i * 3 + 2] = z;
         if (fp64) {
-          nextPositions64xyLow[nextI * 2] = xLow;
-          nextPositions64xyLow[nextI * 2 + 1] = yLow;
+          positions64xyLow[i * 2] = fp64LowPart(x);
+          positions64xyLow[i * 2 + 1] = fp64LowPart(y);
         }
-        nextI++;
-      }
-      if (vertexIndex === 0) {
-        popStartVertex();
-        pushStartVertex(x, y, z, xLow, yLow);
-      }
+        i++;
+      });
+      /* We are reusing the some buffer for `nextPositions` by offsetting one vertex
```

Comment:
The word `enabled` makes it sounds like it is something we can control dynamically, as if it is there to support a feature, not just mark dummy verts.

Ideas: `vUnusedVertex`, `vertexValid`, `fillerVertex` ...?